airbyte-integrations/connectors/source-amazon-ads/integration_tests/abnormal_state.json

```diff
@@ -1,7 +1,15 @@
-{
-  "sponsored_products_report_stream": {
-    "1861552880916640": {
-      "reportDate": "20990101"
+[
+  {
+    "type": "STREAM",
+    "stream": {
+      "stream_state": {
+        "1861552880916640": {
+          "reportDate": "20990101"
+        }
+      },
+      "stream_descriptor": {
+        "name": "sponsored_products_report_stream"
+      }
     }
   }
-}
+]
```

Comment:
I'm under the impression that legacy state format does not prevent SAT from running successfully because the migrated connectors had their SAT pass. I think to avoid the kind of mess we're currently dealing with it would have been better to:
* Have all our connectors setup.py set `airbyte-cdk~=0.1`
* Release our new cdk upgrade as a major version 
* Manually update setup.py of connectors we want to migrate

It would have given us more control. Nevertheless, the per-stream state migration should not be a breaking change (except for the bug you fixed).